src/Build/Graph/ProjectGraph.cs

```diff
@@ -169,17 +172,19 @@ public ProjectGraph(IEnumerable<ProjectGraphEntryPoint> entryPoints, ProjectColl
         {
             ErrorUtilities.VerifyThrowArgumentNull(projectCollection, nameof(projectCollection));
 
-            var nodeStates = new Dictionary<ConfigurationMetadata, NodeState>();
+            var nodeStates = new Dictionary<ProjectGraphNode, NodeState>();
             var entryPointNodes = new List<ProjectGraphNode>();
+            var tasksInProgress = new ConcurrentDictionary<ConfigurationMetadata, Task>();
             foreach (var entryPoint in entryPoints)
             {
                 PropertyDictionary<ProjectPropertyInstance> globalPropertyDictionary = CreatePropertyDictionary(entryPoint.GlobalProperties);
                 var configurationMetadata = new ConfigurationMetadata(FileUtilities.NormalizePath(entryPoint.ProjectFile), globalPropertyDictionary);
-                if(!nodeStates.TryGetValue(configurationMetadata, out NodeState nodeState))
+                LoadGraph(configurationMetadata, projectCollection, globalPropertyDictionary, tasksInProgress);
```

Comment:
Can LoadGraph take a list of entrypoints now so the entrypoints can be processed in parallel? Right now it will process one entrypoint in parallel, then the next, etc.